awsbestpractices/batch-write-output-ignored/BatchWriteOutputIgnoredRule.java

```diff
@@ -0,0 +1,123 @@
+        if (!StringUtils.isBlank(queueURL)) {
+            try {
+                int batchSteps = 1;
+                int maxForBatch = numberOfMessages;
+                if ((numberOfMessages > MAX_MESSAGES)) {
+                    batchSteps = (numberOfMessages / MAX_MESSAGES) + ((numberOfMessages % MAX_MESSAGES > 0) ? 1 : 0);
+                    maxForBatch = MAX_MESSAGES;
+                }
+
+                for (int i = 0; i < batchSteps; i++) {
+                    List<Message> list = amazonSQS.receiveMessage(new ReceiveMessageRequest(queueURL).
+                            withMaxNumberOfMessages(maxForBatch).withWaitTimeSeconds(POLLING_INTERVAL)).getMessages();
+                    if (list != null && !list.isEmpty()) {
+                        List<DeleteMessageBatchRequestEntry> del = new ArrayList<>();
+                        for (Message msg : list) {
+                            messages.add(msg.getBody());
+                            del.add(new DeleteMessageBatchRequestEntry(msg.getMessageId(), msg.getReceiptHandle()));
+                        }
+                        amazonSQS.deleteMessageBatch(queueURL, del);
+                    }
+                }
+            } catch (AmazonServiceException ase) {
+                logException(ase);
+            } catch (AmazonClientException ace) {
+                logger.error("Could not reach SQS. {}", ace.toString());
+            }
+        }
+        return messages;
+    }
+
+    // Must be flagged.
+    // Based on
+    // https://github.com/Erudika/para/blob/5d7d776283cc16e370cd493f3070fb591c8ba685/para-server/src/main/java/com/erudika/para/queue/AWSQueueUtils.java#L224
+    private List<String> ignoreOutput_4(final int numberOfMessages, AmazonSQS amazonSQS) {
+        ReceiveMessageRequest receiveRequest = new ReceiveMessageRequest()
+                .withQueueUrl(dlqURL)
+                .withMaxNumberOfMessages(numberOfMessages)
+                .withVisibilityTimeout(2)
+                .withWaitTimeSeconds(0);
+        ReceiveMessageResult receiveMessageResult = amazonSQS.receiveMessage(receiveRequest);
+
+        List<MessagePair> processedMessages = receiveMessageResult.getMessages().stream()
+                .map(this::processMessage)
+                .collect(Collectors.toList());
+
+        if (!processedMessages.isEmpty()) {
+            DeleteMessageBatchRequest deleteMessageBatchRequest = new DeleteMessageBatchRequest()
+                    .withQueueUrl(dlqURL)
+                    .withEntries(processedMessages.stream()
+                            .map(MessagePair::getMessage)
+                            .map(x -> new DeleteMessageBatchRequestEntry()
+                                    .withId(x.getMessageId())
+                                    .withReceiptHandle(x.getReceiptHandle()))
+                            .collect(Collectors.toList()));
+            amazonSQS.deleteMessageBatch(deleteMessageBatchRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code makes a call to a batch operation, but it does not handle any errors returned by that batch operation. Batch operations can succeed without throwing exceptions even if some items in the request fail. Consider checking the return value of the batch operation for failures.